src/linkml_arrays/dumpers/yaml_xarray_dumpers.py

```diff
@@ -0,0 +1,63 @@
+"""Class for dumping a LinkML model to YAML with paths to NumPy files."""
+
+from pathlib import Path
+from typing import List, Union
+
+import numpy as np
+import xarray as xr
+
+from .yaml_array_file_dumper import YamlArrayFileDumper
+
+
+class YamlXarrayNetCDFDumper(YamlArrayFileDumper):
+    """Dumper class for LinkML models to YAML with paths to .nc file.
+
+    Each array is written to a netcdf dataset at path "/data" in a new .nc file.
+    """
+
+    FILE_SUFFIX = ".nc"  # used in parent class
+    FORMAT = "netcdf"
+
+    @classmethod
+    def write_array(
+        cls, array: Union[List, np.ndarray], output_file_path_no_suffix: Union[str, Path]
+    ):
+        """Write an array to a NumPy file."""
+        # TODO do not assume that there is only one by this name
+        # add suffix to the file name
+        if isinstance(output_file_path_no_suffix, str):
+            output_file_path_no_suffix = Path(output_file_path_no_suffix)
+        output_file_path = output_file_path_no_suffix.parent / (
+            output_file_path_no_suffix.name + cls.FILE_SUFFIX
+        )
+
+        data_array = xr.DataArray(data=array)
+        data_array.to_netcdf(output_file_path, engine="h5netcdf")
+        return output_file_path
+
+
+class YamlXarrayZarrDumper(YamlArrayFileDumper):
```

Comment:
I am sorta having a tough time telling whats going on in this PR (sorry for missing the meeting, I never know when they are), but im doing this rn and u will definitely want to have a model for this, otherwise all the ser/des logic gets packed into a huge nesty dumper/loader and u lose the declarative relationship between in memory and dumped.